docs/learn/dapp-developer/set-nft-metadata.md

```diff
@@ -0,0 +1,114 @@
+---
+sidebar_label: '🎨 Set LSP8 NFT Metadata'
+sidebar_position: 9
+description: Learn how to set the metadata of a NFT part of a LSP8 Collection.
+---
+
+# Set LSP8 NFT Metadata
+
+In this guide you will learn how to set the metadata of a specific NFT (represented by its `bytes32 tokenId`) part of a [LSP8 Identifiable Digital Asset](../../standards/tokens/LSP8-Identifiable-Digital-Asset.md) collection.
+
+## Setup
+
+You will need the address of an existing LSP8 NFT collection in order to follow this tutorial. This represent the LSP8 collection that includes the NFT you want to update metadata for.
+
+The following code snippets require the installation of the following libraries:
+
+- [`ethers.js`](https://github.com/ethers-io/ethers.js/)
+- [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/)
+
+```shell
+npm install ethers @lukso/lsp-smart-contracts
+```
+
+## Imports and constants
+
+At this point, we are preparing the `LSP8IdentifiableDigitalAsset` contract to interact with it. We will construct an instance of the contract using its _ABI_ and the _contract address_.
+
+```javascript
+import LSP8IdentifiableDigitalAsset from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+import { ethers } from 'ethers';
+
+const privateKey = '0x...';
+const lsp8ContractAddress = '0x...';
+
+const provider = new ethers.JsonRpcProvider('https://rpc.testnet.lukso.network');
+
+// setup signer
+const signer = new ethers.Wallet(privateKey as string, provider);
+console.log('🔑 EOA: ', signer.address);
+```
+
+### Instantiate the LSP8 contract
+
+Create an instance of the LSP8 collection contract as shown below:
+
+```javascript
+const lsp8Contract = new ethers.Contract(lsp8ContractAddress, LSP8IdentifiableDigitalAsset.abi, signer);
+```
+
+### Send the transaction to set metadata for a tokenId
+
+The last step is to set the metadata for a specific `tokenId`. You can then send the transaction to set the metadata for the given NFT represented by its `tokenId`.
+
+Here we will use the following parameters as examples:
+
+- `tokenId`: `0x0000000000000000000000000000000000000000000000000000000000000001`
```

Comment:
does `tokenId` support hex or decimal?

Where can i find the specs about this tokenId value?

Might be nice to add more info/links here.

For token 10 shall I write:

```
0x0000000000000000000000000000000000000000000000000000000000000010
```

or 

```
0x000000000000000000000000000000000000000000000000000000000000000A
```

?

Reply:
It is always a `bytes32` as hex value. We can link to the LSP8 standard page in the docs.

Reply:
I suggest to add it right here. It doesn't hurt

```
Note that the tokenId is a bytes32 value (hex)
```